addons/mail/models/mail_alias.py

```diff
@@ -228,6 +229,47 @@ def _compute_display_name(self):
             else:
                 record.display_name = _("Inactive Alias")
 
+    def _search_display_name(self, operator, operand):
+        """ When searching for 'sales@mycompany.com' we should search on name
+        being sales and domain being mycompany.com. """
+        # mono operand search: check if search on complete email or left/right part
+        if operator in {'=', '!=', '=like', '=ilike', 'like', 'not like', 'ilike', 'not ilike'}:
+            if '@' in (operand or ''):
+                left_part, right_part = operand.split('@', 1)
+                return ['&', ('alias_name', operator, left_part), ('alias_domain_id.name', operator, right_part)]
+            return ['|', ('alias_name', operator, operand), ('alias_domain_id.name', operator, operand)]
```

Comment:
this is not 100% correct, for example if we have an alias `test@odoo.com`, and I search for `('display_name', 'ilike', 'te@doo.com')` it will find the record (because it does the ilike on both left / right part)

what about storing that field to help the search ? (and store False instead of `Inactive Alias` so we don't store translated data)